Rubberduck.Inspections/QuickFixProvider.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Inspections
+{
+    public class QuickFixProvider : IQuickFixProvider
+    {
+        private readonly RubberduckParserState _state;
+        private readonly Dictionary<Type, HashSet<IQuickFix>> _quickFixes = new Dictionary<Type, HashSet<IQuickFix>>();
+
+        public QuickFixProvider(RubberduckParserState state, IEnumerable<IQuickFix> quickFixes)
+        {
+            _state = state;
+            foreach (var quickFix in quickFixes)
+            {
+                foreach (var supportedInspection in quickFix.SupportedInspections)
+                {
+                    if (_quickFixes.ContainsKey(supportedInspection))
+                    {
+                        _quickFixes[supportedInspection].Add(quickFix);
+                    }
+                    else
+                    {
+                        _quickFixes.Add(supportedInspection, new HashSet<IQuickFix> {quickFix});
+                    }
+                }
+            }
+        }
+
+        public IEnumerable<IQuickFix> QuickFixes(IInspectionResult result)
+        {
+            return _quickFixes[result.Inspection.GetType()];
+        }
+
+        public void Fix(IQuickFix fix, IInspectionResult result)
+        {
+            fix.Fix(result);
+            _state.GetRewriter(result.QualifiedSelection.QualifiedName).Rewrite();
```

Comment:
Actually no: like refactorings don't call `Rewrite` themselves either (we need a series of tests for that), so that the caller can do multiple modifications without having to re-parse between each operation.

 Makes me wonder whether `IModuleRewriter.Rewrite()` should be called from here too. Who's the caller here?